src/main/java/org/candlepin/model/AbstractHibernateCurator.java

```diff
@@ -996,24 +616,32 @@ public Collection<E> updateAll(Collection<E> entities, boolean flush, boolean ev
         return entities;
     }
 
+    private boolean isEntityNew(E entity) {
+        return entity.getId() == null;
+    }
```

Comment:
The general logic here is exactly what Hibernate+JPA does as well: check if the entity already has a primary key defined, if so: it is an existing entity, otherwise it is a new entity. We've leveraged it a couple times to work around weird edge cases with JPA throughout CP.

However, to Nikos' point, we don't always use the ID field as the primary key (despite it being what is defined by the `Persisted` interface), so this method won't work for all entities and yet it's extracted and named as if it's generic enough to work.

There are a few ways around this. AHC could define it as a public or protected method and let subclasses override it as necessary. Alternatively, we could use the clunky JPA interface to get it generically as we were before:

```
Object id = this.getEntityManager()
  .getEntityManagerFactory()
  .getPersistenceUnitUtil()
  .getIdentifier(entity);
```

Reply:
done